src/main/webapp/view/discussion/discussion-area.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {timestampToString} from '../../timestamps.js';
+
+import DiscussionManager from './discussion-manager.js';
+import {COMMENT_TYPE_QUESTION} from './discussion.js';
+import {DiscussionComment} from './discussion.js';
+
+export const ELEMENT_DISCUSSION =
+    document.querySelector('#discussion-comments');
+
+export default class DiscussionArea {
+  static #ELEMENT_POST_TEXTAREA = document.querySelector('#post-textarea');
+  static #ELEMENT_TIMESTAMP_SPAN = document.querySelector('#timestamp-span');
+  static #TIME_TOLERANCE_MS = 10000;  // 10 seconds.
+  #lecture;
+  #manager;
+  #currentTimeMs;
+  #currentRootCommentElements;
+
+  constructor(lecture) {
+    this.#lecture = lecture;
+    this.#manager = new DiscussionManager(this.#lecture);
+    this.#currentTimeMs = 0;
+    this.#currentRootCommentElements = [];
```

Comment:
Would it make sense to initialize in the constructor?

Reply:
No, that requires a fetch and does a lot of work.